### ISSUE TITLE
fix(core): empty .config.json file in project root gets picked up as lerna-lite config

### ISSUE THREAD

**Issue by petermetz** [lerna-lite/lerna-lite#729]
### Describe the bug

If I have an empty `.config.json` file in my project root then lerna-lite will crash because it assumes that the config file was for itself and not for something else (and it tries to parse it and then crash because the config property is undefined)

IMO, since lerna-lite is a tool that is commonly used as part of tooling of projects that have their own config files, the tool shouldn't assume that a generic name such as .config.json is a config file that is intended for itself (now that I think about it my app also shouldn't assume that TBH, but maybe it has a slightly stronger claim to "squat" on a generic name like this).

### Expectation

TLDR: The default search path for config files should consist of files that have lerna or lerna-lite in their names, not the generic ones. This is just my opinion though, not a strongly held one either.

Or if you think that it's okay to assume `.config.json` belonging to lerna-lite, that's fine by me, but then it should at least validate the config and error out with something more obvious explaining that the config was missing certain (mandatory) properties from it.

### Reproduction

```sh

cd some-lerna-project-root-dir
touch .config.json
yarn lerna run 'echo hello'
```

Which will then crash with:

```sh
TypeError: Cannot read properties of undefined (reading 'rootPath')
    at file:///home/peter/a/cacti/node_modules/@lerna-lite/core/dist/command.js:51:47
```

And if you add some more logging inside the error handler, you can log out this one in addition (same crash):

```sh

TypeError: Cannot use 'in' operator to search for 'extends' in undefined
    at applyExtends (file:///*****/node_modules/@lerna-lite/core/dist/project/lib/apply-extends.js:9:19)
    at Object.transform (file:///*****/node_modules/@lerna-lite/core/dist/project/project.js:47:34)
    at load (/*****/node_modules/cosmiconfig/dist/ExplorerSync.js:20:32)
    at emplace (/*****/node_modules/cosmiconfig/dist/util.js:12:20)
    at ExplorerSync.load (/*****/node_modules/cosmiconfig/dist/ExplorerSync.js:23:42)
    at ExplorerSync.search (/*****/node_modules/cosmiconfig/dist/ExplorerSync.js:30:33)
    at new Project (file:///*****/node_modules/@lerna-lite/core/dist/project/project.js:60:31)
    at file:///*****/node_modules/@lerna-lite/core/dist/command.js:30:54
file:///*****/node_modules/@lerna-lite/core/dist/command.js:51
                    writeLogFile(this.project.rootPath);
                                              ^
```

### Lerna config and logs

<details><summary>lerna.json</summary><p>

```json
{
  "packages": [
    "packages/cactus-*",
    "examples/cactus-*",
    "extensions/cactus-*",
    "packages/cacti-*",
    "examples/cacti-*",
    "extensions/cacti-*",
    "weaver/common/protos-js",
    "weaver/sdks/fabric/interoperation-node-sdk",
    "weaver/sdks/besu/node",
    "weaver/core/drivers/fabric-driver",
    "weaver/core/identity-management/iin-agent",
    "weaver/samples/fabric/fabric-cli",
    "weaver/samples/besu/besu-cli",
    "weaver/samples/besu/simpleasset",
    "weaver/samples/besu/simplestate"
  ],
  "version": "2.0.0-alpha.1",
  "npmClient": "yarn",
  "useWorkspaces": "true",
  "command": {
    "version": {
      "message": "chore(release): publish"
    }
  },
  "changelogPreset": "angular"
}
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

</p></details>

### Environment Info

    System:
      OS: Linux 5.10 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
      CPU: (24) x64 12th Gen Intel(R) Core(TM) i7-12850HX
      Memory: 11.17 GB / 15.44 GB
      Container: Yes
      Shell: 5.1.16 - /bin/bash
    Binaries:
      Node: 16.20.2 - ~/.nvm/versions/node/v16.20.2/bin/node
      Yarn: 3.6.0 - ~/.nvm/versions/node/v16.20.2/bin/yarn
      npm: 8.19.4 - ~/.nvm/versions/node/v16.20.2/bin/npm
    npmPackages:
      @lerna-lite/cli: 2.5.1 => 2.5.1
      @lerna-lite/exec: 2.5.1 => 2.5.1
      @lerna-lite/list: 2.5.1 => 2.5.1
      @lerna-lite/publish: 2.5.1 => 2.5.1
      @lerna-lite/run: 2.5.1 => 2.5.1
      @lerna-lite/version: 2.5.1 => 2.5.1

### Used Package Manager

yarn >= 2.x

### Validations

- [X] Follow our [Code of Conduct](https://github.com/lerna-lite/lerna-lite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://github.com/lerna-lite/lerna-lite#readme).
- [X] Check that there isn't [already an issue](https://github.com/lerna-lite/lerna-lite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/lerna-lite/lerna-lite/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
I did not have a chance to try it out yet, but I know there was an update in `cosmiconfig` which extends TypeScript config in this [PR](https://github.com/cosmiconfig/cosmiconfig/pull/311) on that project, I wouldn't be surprised that this caused your problem. Could you try an older version of Lerna-Lite v2.4.0 and see if that fixes your issue to avoid including this newer cosmiconfig version and let me know if that worked.

**Comment by petermetz** [comment]
> I did not have a chance to try it out yet, but I know there was an update in `cosmiconfig` which extends TypeScript config in this [PR](https://github.com/cosmiconfig/cosmiconfig/pull/311) on that project, I wouldn't be surprised that this caused your problem. Could you try an older version of Lerna-Lite v2.4.0 and see if that fixes your issue to avoid including this newer cosmiconfig version and let me know if that worked.
> 
> EDIT
> 
> I think I found pretty much the same error you got in this upstream cosmiconfig issue [Config search fails if there is a file named `.config`](https://github.com/cosmiconfig/cosmiconfig/issues/313), it says the bug was fixed in cosmiconfig v8.3.6 but Lerna-Lite v2.5.1 is referencing cosmiconfig v8.3.5 which is probably why it included similar bug. I will test it out in the next couple days and probably release a new version including this cosmiconfig fix

@ghiscoding Thank you so much for investigating! Looking forward to the fix! If there's anything else I can help with, just let me know!

**Comment by ghiscoding** [comment]
you can maybe verify the cosmiconfig fix by using yarn override `resolutions` to bump cosmiconfig to v8.3.6

**Comment by petermetz** [comment]
> you can maybe verify the cosmiconfig fix by using yarn override `resolutions` to bump cosmiconfig to v8.3.6

@ghiscoding No luck unfortunately. It fails the exact same way with the same repro steps.
I double checked on the file-system that it is importing v8.3.6 and no other version of cosmiconfig is installed dangling either, so I'm 100% that v8.3.6 is still broken. :-(

**Comment by ghiscoding** [comment]
I have opened an issue [Lerna-Lite project having an empty `.config.json` file gets picked up by cosmiconfig and is unexpected - behavior changed after 8.0](https://github.com/cosmiconfig/cosmiconfig/issues/329) in the upstream repo. Waiting for a reply and potential fix, typically resolved within few days or couple weeks, we'll wait and see

A potential alternative might be to use yarn `resolutions` to use a lower version of cosmiconfig (8.0), `"cosmiconfig": "8.0.0"` seems to be the last known working version, that is after trying it in Lerna-Lite with a breakpoint.

**Comment by ghiscoding** [comment]
this should be fixed in [v3.1.0](https://github.com/lerna-lite/lerna-lite/releases/tag/v3.1.0), cheers
